PrototypeDesignPattern1/Rectangle.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+namespace PrototypeDesignPattern1
+{
+    internal class Rectangle : IComponent
+    {
+        private readonly double m_length;
+        private readonly double m_width;
+        private readonly double m_area;
+
+        //constructor
+        public Rectangle(double length, double width)
+        {
+            m_length = length;
+            m_width = width;
+            m_area = m_length * m_length;
+        }
+
+        // clone method
+        public IComponent Clone()
+        {
+            Rectangle clonedRectangle = new Rectangle(m_length, m_width);
```

Comment:
Update it, for example Rectangle is a collection of lines, and you can try by cloning the lines too.

Reply:
Updated code